src/lib/query/iface.rs

```diff
@@ -568,3 +573,40 @@ pub(crate) fn fill_bridge_vlan_info(
     }
     Ok(())
 }
+
+// Currently there is no valid netlink way to get the driver information as the
+// ETHTOOL_GDRVINFO ioctl command has no netlink equivalent. We use sysfs content
+// /sys/class/net/<if_name>/device/ and extract the last element from the "driver"-link
+// (https://docs.kernel.org/admin-guide/sysfs-rules.html)
+fn _get_iface_driver(if_name: String) -> Option<String> {
+    let sysfs_path = format!("/sys/class/net/{if_name}/device/driver");
+    let path = Path::new(&*sysfs_path);
+
+    let full_path = match path.canonicalize() {
+        Ok(i) => i,
+        Err(_e) => {
+            log::info!("No driver for iface {if_name}");
+            return None;
+        }
+    };
+
+    let driver = match full_path.file_name() {
+        Some(i) => i,
+        None => {
+            log::error!("Failed to get driver for iface {if_name}");
```

Comment:
This log carry no additional information, I can already tell something is wrong when got `driver: None`. And this log says things I already know.

Reply:
Fair enough, removed